### PR TITLE
[5.x] Improve ncss error handling and service description

### PR DESCRIPTION
**Pull request by tdrwenski**
- Improve /ncss/point error handling when dataset cannot be opened as a point dataset. Previously, nothing was logged and status was OK. Now the status returned is "method not allowed" and the error log is printed. I also added a test for this.

- Now the description shown in the browser for /ncss/grid or /ncss/point contains the word "grid" or "point" so that you can differentiate them if both are configured. It is a bit of a hacky fix, but they both correspond to the same ServiceType enum in Netcdf-java, and I thought it might not be a good to change that.

---
This change is [Reviewable](https://reviewable.io/reviews/unidata/tds/251)

### CODE REVIEW

File: tds/src/main/java/thredds/server/TdsErrorHandling.java [hunk 1]
```diff
@@ -104,6 +104,16 @@ public ResponseEntity<String> handle(IOException ex) {
...
 
+  @ExceptionHandler(UnsupportedOperationException.class)
+  public ResponseEntity<String> handle(UnsupportedOperationException ex) {
+    logger.warn("TDS Error", ex);
+
+    HttpHeaders responseHeaders = new HttpHeaders();
+    responseHeaders.setContentType(MediaType.TEXT_PLAIN);
+    return new ResponseEntity<>("UnsupportedOperationException: " + htmlEscape(ex.getMessage()), responseHeaders,
+        HttpStatus.METHOD_NOT_ALLOWED);
+  }
```
Comment (haileyajohnson) [inline tds/src/main/java/thredds/server/TdsErrorHandling.java:115]: I'm not sure what the best status code to return here is, I think `method_not_allowed` usually means trying to do the wrong action (e.g. PUT, GET, DELETE) on a resource.

TBH I'd like to go with [this one](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418)

Comment (tdrwenski) [reply tds/src/main/java/thredds/server/TdsErrorHandling.java:115]: yeah I think you are right... could always do 400 but I think that is more for malformed requests?

>[ 400 Bad Request](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)
The server cannot or will not process the request due to an apparent client error (e.g., malformed request syntax, size too large, invalid request message framing, or deceptive request routing)

Comment (haileyajohnson) [reply tds/src/main/java/thredds/server/TdsErrorHandling.java:115]: I think 400 with a good response message would be ok?

Comment (haileyajohnson) [reply tds/src/main/java/thredds/server/TdsErrorHandling.java:115]: I would say _technically_ requesting the wrong type of subset, in a RESTful API, is considered a bad resource, which would be 400

Comment (tdrwenski) [reply tds/src/main/java/thredds/server/TdsErrorHandling.java:115]: In this case the syntax is correct but the dataset cannot be used with the point service. What do you think about:

> [422 Unprocessable Entity](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422)
> 
> The HyperText Transfer Protocol (HTTP) 422 Unprocessable Entity response status code indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.

Comment (haileyajohnson) [reply tds/src/main/java/thredds/server/TdsErrorHandling.java:115]: I'm seeing suggestions for both, so yeah that could work: https://stackoverflow.com/questions/6123425/rest-response-code-for-invalid-data

**Review by haileyajohnson — APPROVED** [review]

**Pull request merged**